corehq/apps/ota/views.py

```diff
@@ -129,14 +129,22 @@ def app_aware_search(request, domain, app_id):
     except CaseSearchUserError as e:
         return HttpResponse(str(e), status=400)
     fixtures = CaseDBFixture(cases).fixture
-    end_time = datetime.now()
+    _log_search_timing(start_time, request, domain)
+    return HttpResponse(fixtures, content_type="text/xml; charset=utf-8")
+
+
+def _log_search_timing(start_time, request, domain):
+    elapsed = (datetime.now() - start_time).total_seconds()
     metrics_histogram("commcare.app_aware_search.processing_time",
-                      int((end_time - start_time).total_seconds() * 1000),
+                      int(elapsed * 1000),
                       bucket_tag='duration_bucket',
                       buckets=(500, 1000, 5000),
                       bucket_unit='ms',
-                      tags={'domain': domain})
-    return HttpResponse(fixtures, content_type="text/xml; charset=utf-8")
+                      tags={'domain': limit_domains(domain)})
+    if elapsed >= 10 and limit_domains(domain) != "__other__":
+        notify_exception(request, "LongCaseSearchRequest", details={
+            'request_dict': dict((request.GET if request.method == 'GET' else request.POST).lists()),
```

Comment:
I'm not certain how sentry truncation works, but my hope is that putting this here will mean I'll get to see the whole thing in logs